swarm/storage/resource.go

```diff
@@ -407,18 +407,38 @@ func (self *ResourceHandler) updateResourceIndex(rsrc *resource, chunk *Chunk, n
 	}
 
 	// update our rsrcs entry map
-	rsrc.lastBlock = nextblock
+	period, version, _, data, err := parseUpdate(chunk.SData[signatureLength:])
+	rsrc.lastPeriod = period
 	rsrc.version = version
-	rsrc.data = make([]byte, len(chunk.SData)-signatureLength)
 	rsrc.updated = time.Now()
-	copy(rsrc.data, chunk.SData[signatureLength:])
-	log.Debug("Resource synced", "name", rsrc.name, "key", chunk.Key, "block", nextblock, "version", version)
+	rsrc.data = make([]byte, len(data))
+	copy(rsrc.data, data)
+	log.Debug("Resource synced", "name", rsrc.name, "key", chunk.Key, "period", rsrc.lastPeriod, "version", rsrc.version)
 	self.resourceLock.Lock()
```

Comment:
There are other places where you lock resourceLock just to read it, wouldn't it be enough to only RLock it in those cases (and use a RWMutex)?

Also for the sake of readability I would extract these to something like:
```
func (self *ResourceHandler) getResource(index string) *resource {
    self.resourceLock.RLock()
    defer self.resourceLock.RUnlock()
    return self.resource[index]
}
```

And a setter to for the write.

This could make the reading and writing oneliner, it would increase readability and safety (harder to forget the mutex). Or this solution is not go-like maybe?